### PR TITLE
Enhance User API: Update endpoint and logic

### PR DESCRIPTION
**Pull request by MikeMoudry**

### CODE REVIEW

**Comment by Metemi** [comment]
Look good, I think :)

**Review by Metemi — APPROVED** [review]

File: Backend-API/src/main/java/com/avangers/backendapi/services/UserService.java [hunk 1]
```diff
@@ -6,4 +6,6 @@
 
 public interface UserService extends UserDetailsService {
   RegisterUserResponseDTO addUser(RegisterUserRequestDTO registerUserRequestDTO);
+  RegisterUserResponseDTO updateUser(Long userId, RegisterUserRequestDTO updateUserRequestDTO);
```
Comment (Normelik) [inline Backend-API/src/main/java/com/avangers/backendapi/services/UserService.java:9]: maybe it is better to use different DTO, not register

File: Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java [hunk 2]
```diff
@@ -33,4 +30,16 @@ public ResponseEntity<?> register(@Valid @RequestBody RegisterUserRequestDTO reg
       return new ResponseEntity<>(error, HttpStatus.BAD_REQUEST);
     }
   }
+
+  @PutMapping("/api/users/{userId}")
```
Comment (Normelik) [inline Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java:34]: we'll only update the logged-in user, right? @markkovari is it possible to use Principal inside parameters?

File: Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java [hunk 3]
```diff
@@ -33,4 +30,16 @@ public ResponseEntity<?> register(@Valid @RequestBody RegisterUserRequestDTO reg
...
+  public ResponseEntity<?> updateUser(@PathVariable Long userId, @Valid @RequestBody RegisterUserRequestDTO updateUserRequestDTO) {
+    try {
+      RegisterUserResponseDTO response = userService.updateUser(userId, updateUserRequestDTO);
+      return new ResponseEntity<>(response, HttpStatus.OK);
```
Comment (Normelik) [inline Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java:38]: >
```suggestion
      return new ResponseEntity<>(userService.updateUser(userId, updateUserRequestDTO), HttpStatus.OK);
```

File: Backend-API/src/main/java/com/avangers/backendapi/services/UserServiceImpl.java [hunk 4]
```diff
@@ -35,4 +35,22 @@ public RegisterUserResponseDTO addUser(RegisterUserRequestDTO registerUserReques
   public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
     return userRepository.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException("Email is not in database"));
   }
+
+  @Override //updating the user data
+  public RegisterUserResponseDTO updateUser(Long userId, RegisterUserRequestDTO updateUserRequestDTO) {
+    User existingUser = userRepository.findById(userId)
+            .orElseThrow(() -> new IllegalArgumentException("User not found"));
```
Comment (Normelik) [inline Backend-API/src/main/java/com/avangers/backendapi/services/UserServiceImpl.java:42]: UsernameNotFoundException?

**Review by Normelik — COMMENTED** [review]
few recommendations, but overall nice code

File: Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java (hunk 2 above)
Comment (markkovari) [reply Backend-API/src/main/java/com/avangers/backendapi/controllers/UserController.java:34]: Yesyes, this is the way 
image

File: Backend-API/src/main/java/com/avangers/backendapi/services/UserServiceImpl.java (hunk 4 above)
Comment (markkovari) [reply Backend-API/src/main/java/com/avangers/backendapi/services/UserServiceImpl.java:42]: Oh, yes that would be way better, I was just going with the recommendations of IntelliJ/copilot. You are correct, or we can write one of our own.

**Review by markkovari — APPROVED** [review]

File: Backend-API/src/main/java/com/avangers/backendapi/services/UserService.java (hunk 1 above)
Comment (ReneVajda1) [reply Backend-API/src/main/java/com/avangers/backendapi/services/UserService.java:9]:
```suggestion
 UpdateUserResponseDTO updateUser(Long userId, UpdateUserRequestDTO updateUserRequestDTO);
```

maybe < ?

**Review by ReneVajda1 — APPROVED** [review]
+

**Pull request merged**